activerecord/test/cases/invertible_migration_test.rb

```diff
@@ -345,6 +353,28 @@ def test_migrate_down_with_table_name_prefix
       ActiveRecord::Base.table_name_prefix = ActiveRecord::Base.table_name_suffix = ''
     end
 
+    # Nested transactions in migrations don't work with MySQL
+    unless current_adapter?(:MysqlAdapter, :Mysql2Adapter)
+      def test_migrate_revert_transaction_by_part
```

Comment:
This test-case looks like a very close copy of the test without a transaction wrapped around. Can we reuse the test-logic instead of duplicating the whole thing?